app/api/contact/specs/routes.spec.js

```diff
@@ -1,37 +1,54 @@
-import { catchErrors } from 'api/utils/jasmineHelpers';
-import conctactRoutes from '../routes.js';
-import instrumentRoutes from '../../utils/instrumentRoutes';
+import request from 'supertest';
+import { setUpApp } from 'api/utils/testingRoutes';
+import contactRoutes from '../routes.js';
 import contact from '../contact';
 
-describe('entities', () => {
-  let routes;
+jest.mock('../../utils/languageMiddleware.ts', () => (_req, _res, next) => {
+  next();
+});
 
-  beforeEach(() => {
-    routes = instrumentRoutes(conctactRoutes);
-  });
+describe('contact', () => {
+  const app = setUpApp(contactRoutes);
 
   describe('POST', () => {
-    let req;
+    let body;
     beforeEach(() => {
-      req = {
-        body: { name: 'Bruce Wayne', email: 'notbatman@wayne.com', text: 'I want to donate!' },
-        language: 'lang',
-      };
+      body = { name: 'Bruce Wayne', email: 'notbatman@wayne.com', message: 'I want to donate!' };
     });
 
-    it('should have a validation schema', () => {
-      expect(routes.post.validation('/api/contact')).toMatchSnapshot();
+    it('should send an email', async () => {
+      spyOn(contact, 'sendMessage').and.returnValue(Promise.resolve());
+      const response = await request(app)
+        .post('/api/contact')
+        .send(body);
+      expect(response.text).toContain('ok');
+      expect(contact.sendMessage).toHaveBeenCalledWith(body);
     });
 
-    it('should send an email', done => {
-      spyOn(contact, 'sendMessage').and.returnValue(Promise.resolve());
-      routes
-        .post('/api/contact', req)
-        .then(() => {
-          expect(contact.sendMessage).toHaveBeenCalledWith(req.body);
-          done();
-        })
-        .catch(catchErrors(done));
+    it('should next with error if sending email failed', async () => {
+      spyOn(contact, 'sendMessage').and.callFake(() => Promise.reject());
+      const response = await request(app)
+        .post('/api/contact')
+        .send(body);
+      expect(response.status).not.toBe(200);
+    });
+
+    describe('validation', () => {
+      it('should not validate with missing requried properties properties', async () => {
+        body = {};
+        const response = await request(app)
+          .post('/api/contact')
+          .send(body);
+        expect(response.text).toContain('validation failed');
+      });
+
+      it('e-mail should be an e-mail', async () => {
+        body.email = 'wrong format';
+        const response = await request(app)
+          .post('/api/contact')
+          .send(body);
+        expect(response.text).toContain('validation failed');
```

Comment:
This is probably the wrong thing to expect.  Since you are missing the name and message, this test will pass even if you include no mail at all.  Or even if you put in a correct mail in the body!

So, there should be a specific failure, on top of the validation failed, that proves that email is actually being rejected.  For such a scenario, it's probably a good practice to include the other fields passing correctly and just expect the specific mail error you are expecting from AJV.

Reply:
Actually is being tested as you suggest. body is set on beforeEach with correct params. I'm only overwriting e-mail with a wrong param.

Reply:
@txau but the test should expect that the error is an email error instead of a generic ajv Error i think that is what @RafaPolit is pointing out.

we already have this scenarios when testing entity/template validators and the tests were actually green when the error had nothing to do with the specific test but for another unexpected scenario, its true that is a much simpler scenario...

Reply:
@daneryl still it will be coupled to how AJV reports the errors.

Reply:
@txau im ok in this case but at some point i think it is more important to test what is actually being validated and be coupled to whatever you are using as a validator, we are coupled already anyways.